### PR TITLE
cnn_3dobj: fix include and allow compilation of samples

### LINKED ISSUE #580: Error on compiling: caffe/vision_layers.hpp: No such file or directory
I am getting compilation error 

In file included from /home/jason/opencv_contrib/modules/cnn_3dobj/src/precomp.hpp:45:0,
                 from /home/jason/opencv/build/modules/cnn_3dobj/opencv_cnn_3dobj_pch_dephelp.cxx:1:
/home/jason/opencv_contrib/modules/cnn_3dobj/include/opencv2/cnn_3dobj.hpp:68:35: fatal error: caffe/vision_layers.hpp: No such file or directory
 #include <caffe/vision_layers.hpp>
                                   ^
compilation terminated.

I checked the master branch of caffe https://github.com/BVLC/caffe, and no vision_layers.hpp exist. Please look into this. Thanks!

Edit: I am on Ubuntu 14.04

### PR DESCRIPTION
**Pull request by paroj**
caffe/vision_layers.hpp is gone. also fix some warnings.

fixes #580

### CODE REVIEW

**Comment by sovrasov** [comment]
@paroj please use resolves instead of fixes in the first comment (in this case the corresponding issue will be closed by buildbot when PR will be merged) and use **WIP** keyword if you are planning to add more commits into the PR.

**Comment by alalek** [comment]
> resolves instead of fixes

Actually both are valid for GitHub: https://help.github.com/articles/closing-issues-via-commit-messages.
But please don't use this functionality in commits directly (PR description is the optimal way).

`**WIP**` in description doesn't run automatically any builds (this may be a problem because there is no access to rerun any builds). After remowing WIP you need to push new commits (or to amend previous commint and force push) to run automatic builds again.

**Comment by paroj** [comment]
@sovrasov this is not WIP. I just tend to add commits on existing PRs instead of opening new ones.

File: modules/cnn_3dobj/test/test_cnn_3dobj_feature_extract.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@ void CV_CNN_Feature_Test::run(int)
     String network_forIMG = cvtest::TS::ptr()->get_data_path() + "3d_triplet_testIMG.prototxt";
     String mean_file = "no";
     std::vector<String> ref_img;
-    String target_img = String(ts->get_data_path()) + "1_8.png";
+    String target_img = String(ts->get_data_path()) + "4_78.png";
```
Comment (sovrasov) [inline modules/cnn_3dobj/test/test_cnn_3dobj_feature_extract.cpp]: I have an error here:
`could not read reference image /home/user/repositories/opencv_extra/testdata/cv/4_78.png`

Comment (paroj) [inline modules/cnn_3dobj/test/test_cnn_3dobj_feature_extract.cpp]: currently, you still have to set
`export OPENCV_TEST_DATA_PATH=<opencv_contrib>/modules/cnn_3dobj/testdata`

see: https://github.com/opencv/opencv_contrib/tree/master/modules/cnn_3dobj

Comment (sovrasov) [inline modules/cnn_3dobj/test/test_cnn_3dobj_feature_extract.cpp]: Ok, lets wait for https://github.com/opencv/opencv/pull/7750. It allows to add different testdata sources using `ocv_add_testdata` macro in CMake.

Comment (paroj) [inline modules/cnn_3dobj/test/test_cnn_3dobj_feature_extract.cpp]: done

Comment (alalek) [inline modules/cnn_3dobj/test/test_cnn_3dobj_feature_extract.cpp]: @paroj Please apply these [changes](https://github.com/alalek/opencv_contrib/commit/pr871).
After that you can put `OPENCV_TEST_DATA_PATH` back to `<opencv_extra>/testdata`

**Comment by alalek** [comment]
Thank you! :+1:

**Comment by mshabunin** [comment]
@paroj , can you please confirm whether we need changes from #453

**Pull request merged**